docs/tutorial.md

```diff
@@ -0,0 +1,369 @@
+                        .deployments()
+                        .inNamespace(resource.getMetadata().getNamespace())
+                        .withName(resource.getMetadata().getName())
+                        .get();
+```
+
+If the deployment is `null` that means we need to create the deployment for it.
+
+Below code will verify that Deployment within the cluster got created or not. If deployment is null then it will create deployment. `createMemcachedDeployment(resource)` creates the Deployment and then it will be applied by using `client.apps().deployments().create(newDeployment);` code. `createMemcachedDeployment(resource)` method explained in the next part.
+
+```
+        if (deployment == null) {
+            Deployment newDeployment = createMemcachedDeployment(resource);
+            client.apps().deployments().create(newDeployment);
+            return UpdateControl.noUpdate();
+        }
+```
+
+Once we create the deployment, we need to decide whether we have to reconcile it or not.
+If there is no need of reconciliation then return `UpdateControl.noUpdate()` else we need to return `UpdateControl.updateStatusSubResource(resource)`
+
+After the Creation of the Deployment, get the current and required replicas by using the below code.
+
+```
+        int currentReplicas = deployment.getSpec().getReplicas();
+        int requiredReplicas = resource.getSpec().getSize();
+```
+
+If currentReplicas does not match with requiredReplicas then we need to update the `Deployment`. This process will be done by using the below code.
+
+```
+        if (currentReplicas != requiredReplicas) {
+            deployment.getSpec().setReplicas(requiredReplicas);
+            client.apps().deployments().createOrReplace(deployment);
+            return UpdateControl.noUpdate();
+        }
+```
+
+Then, get the list of pods and pod names with the below code.
+
+```
+        List<Pod> pods =
+                client
+                        .pods()
+                        .inNamespace(resource.getMetadata().getNamespace())
+                        .withLabels(labelsForMemcached(resource))
+                        .list()
+                        .getItems();
+
+        List<String> podNames =
+                pods.stream().map(p -> p.getMetadata().getName()).collect(Collectors.toList());
+```
+
+Now, check whether resources get created or not. Then, it verifies podnames with the Memcached resources. If there is a mismatch in either of these conditions then we need to do a reconciliation.
+
+```
+        if (resource.getStatus() == null
+                || !CollectionUtils.isEqualCollection(podNames, resource.getStatus().getNodes())) {
+            if (resource.getStatus() == null) resource.setStatus(new MemcachedStatus());
+            resource.getStatus().setNodes(podNames);
+            return UpdateControl.updateStatusSubResource(resource);
+        }
+```
+
+### createMemcachedDeployment
+
+Thie method simply creates the Deployment as specified below.
```

Comment:
What is this and where does it belong? Is this something I have to create?

Reply:
Modified it in the intro.